src/main/java/com/exasol/errorcodecrawlermavenplugin/ErrorCodeCrawlerMojo.java

```diff
@@ -57,7 +58,9 @@ public class ErrorCodeCrawlerMojo extends AbstractMojo {
     // [impl->dsn~src-directory-override]
     private List<Path> getSourcePaths() {
         if (this.sourcePaths == null || this.sourcePaths.isEmpty()) {
-            return List.of(Path.of("src/main/java"));
+            final var projectDir = this.project.getBasedir().toPath();
+            final var rootProjectDir = this.project.getParent() == null ? projectDir : this.project.getParent().getBasedir().toPath();
+            return List.of(Path.of(rootProjectDir.relativize(projectDir).toString(), "src", "main", "java"));
```

Comment:
Is it necessary to use the root project dir also here? I think we only need it in the `execute()` method.

Reply:
Yes, the original code used Path.of("src/main/java"), which was not correct for sub-projects. So, in order to fix the bug, we should also fix the code here.
I have improved the code a little bit, by passing projectDir and rootProjectDir as parameters from execute method.

Reply:
Instead of rootProjectDir.relativize(projectDir) we might also use projectDir.getFile().getName + "/src/main/java", but in general, it will not work for nested sub-projects with nesting level more than one. So I think rootProjectDir.relativize(projectDir) is generally safer to use.